File: background.py
import pygame
from pygame.math import Vector2
import random

w = 1000
h = 1000
background = "black"

color = "white"
size = 2.5
vel = 1
decimals = 2
maximal = False

count = 100

class Blob:
    def __init__(self, size, maxvel, color, max = False):
        self.size = size
        self.pos = self.randpos()
        if max:
            self.vel = Vector2(maxvel, 0)
        else:
            self.vel = self.randvel(maxvel)
        self.color = color
        
    def randpos(self):
        x = random.randrange(0, w)
        y = random.randrange(0, h)
        
        return Vector2(x, y)
    
    def randvel(self, maxvel):
        factor = 10**decimals
        maxvel = maxvel*factor
        
        minvel = -1*maxvel
        
        xvel = random.randrange(minvel, maxvel)
        xvel = xvel/factor
        
        yvel = random.randrange(minvel, maxvel)
        yvel = yvel/factor
        
        return Vector2(xvel, yvel)
    
    def move(self):
        self.pos += self.vel
        
    def border(self):
        if self.pos.x > w:
            self.pos.x -= w
        
        elif self.pos.x < 0:
            self.pos.x += w
        
        if self.pos.y > h:
            self.pos.y -= h
        
        elif self.pos.y < 0:
            self.pos.y += h
    
    def draw(self):
        pygame.draw.circle(screen, self.color, self.pos, self.size)
        
    def cycle(self):
        self.move()
        self.border()
        self.draw()
    

# pygame setup
pygame.init()
screen = pygame.display.set_mode((w, h), pygame.RESIZABLE)
clock = pygame.time.Clock()
running = True

blobs = []

while len(blobs) < count:
    blob = Blob(size, vel, color, maximal)
    blobs.append(blob)

while running:
    # poll for events
    # pygame.QUIT event means the user clicked X to close your window
    
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
            
        if event.type == pygame.VIDEORESIZE:
            w, h = event.w, event.h
            screen = pygame.display.set_mode((w, h), pygame.RESIZABLE)

    # fill the screen with a color to wipe away anything from last frame
    screen.fill(background)

    # RENDER YOUR GAME HERE
    
    for blob in blobs:
        blob.cycle()

    # flip() the display to put your work on screen
    pygame.display.flip()

    clock.tick(60)  # limits FPS to 60

pygame.quit()

File: background.js
// =========================
// PARAMETERS (easy to change)
// =========================
let w = window.innerWidth;
let h = window.innerHeight;
let background = "rgb(5, 5, 5)";

let blobColor = "rgba(102, 102, 102, 1)";
let blobSize = 2;
let blobVel = 0.2;     // max speed
let count = 200;
let decimals = 2;      // number of decimal places for velocity
let direction = 0.1 * Math.PI;  // angle in radians (null = random per blob)
let sameSpeed = false; // if true, all blobs move with exactly the same speed

// =========================
// HELPER FUNCTIONS
// =========================
function randomVel(maxVel) {
  let minVel = 0.1 * maxVel; // minst 10% av max
  let factor = Math.pow(10, decimals);
  let vel;
  do {
    vel = (Math.floor((Math.random() * 2 * maxVel * factor) - maxVel * factor)) / factor;
  } while (Math.abs(vel) < minVel); // se till att hastigheten inte är för låg
  return vel;
}

function randomSpeed(maxVel) {
  let minSpeed = 0.1 * maxVel; // minst 10% av max
  let factor = Math.pow(10, decimals);
  let speed;
  do {
    speed = Math.floor(Math.random() * maxVel * factor) / factor;
  } while (speed < minSpeed); // se till att hastigheten inte är för låg
  return speed;
}

// =========================
// BLOB CLASS
// =========================
class Blob {
  constructor(size, maxVel, color, direction, sameSpeed) {
    this.size = size;
    this.color = color;
    this.pos = { x: Math.random() * w, y: Math.random() * h };

    if (direction !== null) {
      // Alla går åt samma håll
      let speed = sameSpeed ? maxVel : randomSpeed(maxVel);
      this.vel = { x: speed * Math.cos(direction), y: speed * Math.sin(direction) };
    } else {
      // Slumpad riktning
      let vx, vy;
      do {
        vx = randomVel(maxVel);
        vy = randomVel(maxVel);
      } while (vx === 0 && vy === 0); // se till att inte båda är noll
      this.vel = { x: vx, y: vy };
    }
  }

  move() {
    this.pos.x += this.vel.x;
    this.pos.y += this.vel.y;
    this.border();
  }

  border() {
    if (this.pos.x > w) this.pos.x -= w;
    if (this.pos.x < 0) this.pos.x += w;
    if (this.pos.y > h) this.pos.y -= h;
    if (this.pos.y < 0) this.pos.y += h;
  }

  draw(ctx) {
    ctx.beginPath();
    ctx.arc(this.pos.x, this.pos.y, this.size, 0, Math.PI * 2);
    ctx.fillStyle = this.color;
    ctx.fill();
  }

  cycle(ctx) {
    this.move();
    this.draw(ctx);
  }
}

// =========================
// SETUP CANVAS AND BLOBS
// =========================
const canvas = document.getElementById("blobCanvas");
const ctx = canvas.getContext("2d");
canvas.width = w;
canvas.height = h;

const blobs = [];
for (let i = 0; i < count; i++) {
  blobs.push(new Blob(blobSize, blobVel, blobColor, direction, sameSpeed));
}

// =========================
// ANIMATION LOOP
// =========================
function animate() {
  ctx.fillStyle = background;
  ctx.fillRect(0, 0, w, h);

  blobs.forEach(blob => blob.cycle(ctx));

  requestAnimationFrame(animate);
}

// =========================
// HANDLE RESIZE
// =========================
window.addEventListener('resize', () => {
  w = window.innerWidth;
  h = window.innerHeight;
  canvas.width = w;
  canvas.height = h;
});

animate();
